x-pack/plugins/ml/server/routes/schemas/job_service_schema.ts

```diff
@@ -109,15 +109,11 @@ export const revertModelSnapshotSchema = schema.object({
   ),
 });
 
-export const datafeedPreviewSchema = schema.oneOf([
-  schema.object({
-    job: schema.object(anomalyDetectionJobSchema),
-    datafeed: datafeedConfigSchema,
-  }),
-  schema.object({
-    datafeedId: schema.string(),
-  }),
-]);
+export const datafeedPreviewSchema = schema.object({
+  job: schema.maybe(schema.object(anomalyDetectionJobSchema)),
+  datafeed: schema.maybe(datafeedConfigSchema),
+  datafeedId: schema.maybe(schema.string()),
+});
```

Comment:
I think you need a custom validation callback for this schema because you have mutually exclusive properties.

Reply:
Do you know of an example I can copy?

Reply:
the previous version using `schema.oneOf` i think was ok, but it caused other type errors

Reply:
found this one
https://github.com/elastic/kibana/blob/c388572ece14b031c6b03155e7bff7eae658a80b/x-pack/plugins/ml/server/routes/schemas/alerting_schema.ts#L20

Reply:
@jgowdyelastic yes, something like this

Reply:
added in https://github.com/elastic/kibana/pull/101780/commits/f895bb13866938e2af7d492fdf40c5b4e79e772c